src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ConversationService.java

```diff
@@ -311,16 +311,19 @@ private void sendToConversationMembershipChannel(MetisCrudAction metisCrudAction
     /**
      * Checks if a user is a member of a conversation and therefore can access it else throws an exception
      *
-     * @param conversationId the id of the conversation
-     * @param user           the user to check
+     * @param conversation the conversation the user wants to interact with
+     * @param user         the user to check
      * @return conversation if the user is a member
      */
-    public Conversation mayInteractWithConversationElseThrow(Long conversationId, User user) {
-        Optional<Conversation> conversation = conversationRepository.findById(conversationId);
-        if (conversation.isEmpty() || !isMember(conversationId, user.getId())) {
+    public Conversation mayInteractWithConversationElseThrow(Conversation conversation, User user) {
```

Comment:
I'm not quite sure I understand what you want to accomplish with the changes to this method, except for returning a conversation that also includes participants now.
The call to `findConversationParticipantByConversationIdAndUserIdElseThrow` with the subsequent if statements seems to do the same thing as before, but worse. Before you had a single database call that checked if the participant existed and then only returned a boolean. Now you load the participant from the database instead to do the comparison manually. This is slightly more inefficient for no real benefit IMO.
Also I don't see how a `ConversationParticipant` could exist with the `conversationId`, but not be part of the conversation's `conversationParticipants` set, so the contains call seems to be unnecessary to me. This would only really change the `AccessForbiddenException` to always be an `EntityNotFoundException` now.

If I miss something here, please tell me.